src/lib.rs

```diff
@@ -625,33 +602,98 @@ pub mod liquidity_mining {
     }
 
     #[wasm_bindgen]
-    pub fn calculate_global_farm_shares(valued_shares: String, multiplier: String) -> String {
-        let s = parse_into!(u128, valued_shares);
+    pub fn calculate_yield_farm_rewards(
+        yield_farm_rpz: String,
+        global_farm_rpz: String,
+        multiplier: String,
+        total_valued_shares: String,
+    ) -> String {
+        let y_rpz = FixedU128::from_inner(parse_into!(u128, yield_farm_rpz));
+        let g_rpz = FixedU128::from_inner(parse_into!(u128, global_farm_rpz));
         let m = FixedU128::from_inner(parse_into!(u128, multiplier));
+        let vs = parse_into!(u128, total_valued_shares);
 
-        let result = hydra_dx_math::liquidity_mining::calculate_global_farm_shares(s, m);
+        let result = hydra_dx_math::liquidity_mining::calculate_yield_farm_rewards(y_rpz, g_rpz, m, vs);
 
-        if let Some(r) = result.ok() {
-            r.to_string()
+        if let Some(v) = result.ok() {
+            format!("{},{}", v.0.to_string(), v.1.to_string())
```

Comment:
I would consider splitting into 2 functions ? similar to xyk's remove liquidity ?